src/support/generation.c

```diff
@@ -232,7 +232,15 @@ __gen_oldest(WT_SESSION_IMPL *session, int which)
      * the sessions that could have been active when we started our check.
      */
     WT_ORDERED_READ(session_cnt, conn->session_cnt);
-    for (oldest = conn->generations[which], s = conn->sessions, i = 0; i < session_cnt; ++s, ++i) {
+    /*
+     * On platforms with weaker memory models we need to order the read of the connection generation
+     * before the read of the session generation. If the session generation read is ordered before
```

Comment:
I would omit the mention of weaker memory models, we need this specific ordering, some platforms guarantee it and some don't, I think that's WT_ORDERED_READ's job, to do something or nothing depending on which platform we are building for.

Reply:
[ea517a1](https://github.com/wiredtiger/wiredtiger/pull/9463/commits/ea517a17db053aa41a298195b7daec62607afe4e)